### ISSUE TITLE
[Enhancement] Flag color and font

### ISSUE THREAD

**Issue by henrikvoetmand** [kirbydesign/designsystem#2041]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
To improve readability of text in our red "danger" flags, we would like to change both the background color and the text font.

## Describe the solution you'd like
Set the red background color of "danger" flags to red-30 ( #FF878A - one of the new kirby colors)
Change font weight to "medium" (500) 

## Have you considered any alternatives?
No!

## Are there any additional context?
![Flag issue](https://user-images.githubusercontent.com/71710012/153007481-289c8c21-6882-4e2f-8848-9f3f3f1d1bcd.png)

See [comment](https://github.com/kirbydesign/designsystem/issues/2041#issuecomment-1055502831) for context.
<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [master branch](https://github.com/kirbydesign/designsystem/tree/master) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

**Comment by MadsBuchmann** [comment]
We will use this as a tester for a new design token structure. Therefore we will have a meeting to refine how this should be done. But the task itself should be simple enough - we're just complicating it a bit by testing the new structure 😄

**Comment by MadsBuchmann** [comment]
# Design tokens discussion write up

We should have the following layers: 
1. Named colors (red-30, green-10, etc...)
2. Semantic named tokens (encodes intention, usage & meaning)
3. Application to component (not an actual layer - we just apply the tokens from layer 2 to the individual components)

A naming convention for the second layer could be ([inspired by this talk](https://youtu.be/ylDed18OVdY?t=744)): 

![Screenshot 2022-03-01 at 15 03 57](https://user-images.githubusercontent.com/17269698/156183399-00788c7e-9ba2-4697-8e20-2453b51f30e2.png)

We're still a bit iffy on whether we need each part of that naming convention to serve our needs. 

In the case of flag, we decided on implementing the following structure: 
![IMG_1104](https://user-images.githubusercontent.com/17269698/156186502-d2e2c405-b2a3-4632-801f-aacef09a8cae.jpg)

## What now? 
We discussed whether to include both `danger-background-weak` and `danger-background` for this issue. 

I'm a bit iffy on where we eventually landed. My understanding is, that we decided on introducing `danger-background-weak` for now and then spin off an additional issue for `danger-background` and replacing instances of `red-50` with the new semantic token. 

This is to be able to quickly ship this issue. 

A new issue has been created for this: 
- [ ] Introduce `danger-background` design token

## Generated colors 
@henrikvoetmand mentioned that it is desired to have some colors which are generated programatically to be part of kirby. The code for this is in Sass; we decided that it could be done. A new issue has been created for this: 
- [ ] Include programatically generated colors

**Comment by jkaltoft** [comment]
Ultimately I still think we should go for at ITCSS inspired architecture. To keep the scope of this issue under control I agree that we should "just throw it" into the existing `colors` partial. Later that partial should be placed in the _Settings_ layer. We should probably also hold our horses regarding `style-dictionary`, but it _does_ sound interesting 🙂

**Comment by MadsBuchmann** [comment]
@jkaltoft out of curiosity, if we were to go the ITCSS route the tokens would belong to the settings layer, aight?


![](https://www.xfivecdn.com/xfive/wp-content/uploads/2016/02/01083650/itcss-layers2.svg)

**Comment by jkaltoft** [comment]
> @jkaltoft out of curiosity, if we were to go the ITCSS route the tokens would belong to the settings layer, aight?

Yes! 😁

<img width="1009" alt="image" src="https://user-images.githubusercontent.com/150451/157226821-f4ebd582-157b-4348-bcb1-2bd1b060cb80.png">

**Comment by MadsBuchmann** [comment]
My two remaining brain cells were on vacation when i read that 🤦

**Comment by MadsBuchmann** [comment]
After having discussed this last Friday (11/3/2022), the tokens should just be created in the `_colors.scss` file as described in: https://github.com/kirbydesign/designsystem/issues/2041#issuecomment-1061569887. 

The file structure can then be further refined in the future. For example using ITCSS.
